src/main/ruby/core/string_mirror.rb

```diff
@@ -50,7 +50,13 @@ def character_index(str, start)
       end
 
       def byte_index(value, start=0)
```

Comment:
I'm a bit confused, what does this method do?
Converting a char index to a byte index, or searching for a String and returning its byte index in `@object`?
It would be nice to document this method or use more explicit variable names.

Reply:
It does both. And it is indeed confusing, which is why the duplication probably occurred in the first place. I reworked the primitive and gave it a clearer name. I can rename the method as well. Although I think I'd rather just get rid of the method and inline the body. It's only called in ~6 places and the form to use is fairly obvious.

Reply:
:+1: I think inlining this would make more sense than having a short name for two totally different things.

Reply:
I opted to inline the `String` case and rename the mirror method for now. Inlining the primitive call made the calling code a lot uglier. I think we may want to take a pass over the mirror objects in general though to cut out the indirection. I'll save the remaining work for that pass.